src/renderer/src/pages/settings/settings-general.tsx

```diff
@@ -79,28 +123,40 @@ export function SettingsGeneral({
         </Button>
       </div>
 
+      <h3>{t("language")}</h3>
+      <>
+        <Select value={form.language} onChange={handleLanguageChange}>
+          {languageOptions.map((language) => (
```

Comment:
You can potentially convert this into an `options` prop, since the `Select` children will always be `<option />`
For instance:
```tsx
<Select options={[ { value: "en", label: "English" } ]} />
```